DATA PREPARATION: 

In [1]:
import pandas as pd
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

In [2]:
dataset= pd.read_csv("Employee_Attrition.csv")

In [3]:
dataset.shape

(1470, 35)

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 35 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Age                       1323 non-null   float64
 1   Attrition                 1323 non-null   object 
 2   BusinessTravel            1323 non-null   object 
 3   DailyRate                 1323 non-null   float64
 4   Department                1323 non-null   object 
 5   DistanceFromHome          1323 non-null   float64
 6   Education                 1323 non-null   float64
 7   EducationField            1323 non-null   object 
 8   EmployeeCount             1323 non-null   float64
 9   EmployeeNumber            1323 non-null   float64
 10  EnvironmentSatisfaction   1323 non-null   float64
 11  Gender                    1323 non-null   object 
 12  HourlyRate                1323 non-null   float64
 13  JobInvolvement            1323 non-null   float64
 14  JobLevel

In [5]:
print("The data set has {} rows and {} columns".format(dataset.shape[0], dataset.shape[1]))
display(dataset.describe())
display(dataset.head())
display(dataset.dtypes.value_counts())

The data set has 1470 rows and 35 columns


Age    DailyRate  DistanceFromHome    Education  EmployeeCount  \
count  1323.000000  1323.000000       1323.000000  1323.000000    1323.000000   
mean     36.635268   802.030050          9.091097     2.923447       0.999325   
std       9.882161   414.031025          8.177870     1.072203       0.099656   
min      14.544608    86.828045          0.750112     0.770362       0.628985   
25%      29.624982   456.479553          2.147069     2.122706       0.931146   
50%      34.951950   782.958705          6.920808     3.015658       0.998949   
75%      42.748741  1130.577949         13.584900     3.700560       1.067371   
max      69.402515  1784.394456         33.684424     6.285161       1.290613   

       EmployeeNumber  EnvironmentSatisfaction   HourlyRate  JobInvolvement  \
count     1323.000000              1323.000000  1323.000000     1323.000000   
mean      1026.679166                 2.726443    65.647722        2.743028   
std        610.480802                 1.129679    21.469709        0.765568   
min          1.054148                 0.759655    23.855272        0.722008   
25%        501.854842                 1.894818    47.701897        2.177638   
50%       1022.025017                 2.887313    64.934490        2.859143   
75%       1533.811096                 3.676600    83.038355        3.179128   
max       2530.919375                 5.254672   123.200439        5.241968   

          JobLevel  ...  RelationshipSatisfaction  StandardHours  \
count  1323.000000  ...               1323.000000    1323.000000   
mean      2.058270  ...                  2.720431      80.241313   
std       1.127018  ...                  1.115959       8.022535   
min       0.715172  ...                  0.720696      53.793059   
25%       1.045228  ...                  1.909010      74.882946   
50%       1.936150  ...                  2.891235      80.351400   
75%       2.609707  ...                  3.639961      85.664169   
max       5.942269  ...                  5.092280     107.378705   

       StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
count       1323.000000        1323.000000            1323.000000   
mean           0.814757          11.426152               2.811756   
std            0.863076           8.075870               1.329044   
min            0.000000           0.000000               0.000000   
25%            0.000000           5.953221               1.967328   
50%            0.913651           9.306742               2.690934   
75%            1.082523          15.324548               3.366949   
max            3.474319          47.641892               7.503201   

       WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
count      1323.000000     1323.000000         1323.000000   
mean          2.762825        6.933624            4.233363   
std           0.761802        6.046798            3.658111   
min           0.799138        0.000000            0.000000   
25%           2.193390        2.673269            1.824392   
50%           2.899807        5.422160            2.968874   
75%           3.206117        9.238005            7.052591   
max           4.856086       36.848408           19.091844   

       YearsSinceLastPromotion  YearsWithCurrManager  
count              1323.000000           1323.000000  
mean                  2.160710              4.117695  
std                   3.221481              3.594053  
min                   0.000000              0.000000  
25%                   0.000000              1.687177  
50%                   0.967140              2.939240  
75%                   2.586631              6.987964  
max                  17.273662             19.976003  

[8 rows x 26 columns]

Age Attrition     BusinessTravel    DailyRate  \
0  41.920786       Yes      Travel_Rarely   990.110009   
1  51.589037        No  Travel_Frequently   276.776030   
2  33.131540       Yes      Travel_Rarely  1204.158501   
3  34.707073        No  Travel_Frequently  1352.752432   
4  24.790188        No      Travel_Rarely   485.053333   

               Department  DistanceFromHome  Education EducationField  \
0                   Sales          1.016716   1.770313  Life Sciences   
1  Research & Development               NaN   1.035333  Life Sciences   
2  Research & Development          2.198662   2.118801          Other   
3                   Sales          2.937851   3.670674  Life Sciences   
4  Research & Development          1.988953   1.120496        Medical   

   EmployeeCount  EmployeeNumber  ...  RelationshipSatisfaction StandardHours  \
0       1.033298        1.054148  ...                  1.012832     95.612654   
1       0.944859        2.107031  ...                  4.312008     85.370868   
2       0.810809        4.008125  ...                  2.332558     76.248084   
3       0.822976        4.587269  ...                  3.029591     75.888697   
4            NaN        7.228740  ...                       NaN     72.670937   

   StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear WorkLifeBalance  \
0          0.000000           8.063833               0.000000        1.052104   
1          1.104642           8.965059               3.264952        2.526470   
2          0.000000           8.427238               3.004521        2.924784   
3          0.000000           7.662606               2.856001        3.117053   
4          0.953012           6.241733               2.814719        3.112317   

   YearsAtCompany YearsInCurrentRole  YearsSinceLastPromotion  \
0        5.795945           3.993896                 0.000000   
1       10.070232           7.937505                 1.032295   
2        0.000000           0.000000                 0.000000   
3        8.435324           5.563667                 2.845429   
4        2.040124           1.970786                 2.495949   

   YearsWithCurrManager  
0              4.594972  
1              5.901207  
2              0.000000  
3              0.000000  
4              2.219353  

[5 rows x 35 columns]

float64    26
object      9
Name: count, dtype: int64

It is noticable that the dataset has 1470 observationd and 35 features. Also with the function .info() the dataset shos the present of float and object but no integer and it is noticable thta some of the variable needs to be transformed like for example the Age and more of the variable needs to be transformed in integers

Check for duplicates: 

In [6]:
dataset.duplicated().sum() 

0

Missing values:

In [7]:
dataset.isnull().sum()

Age                         147
Attrition                   147
BusinessTravel              147
DailyRate                   147
Department                  147
DistanceFromHome            147
Education                   147
EducationField              147
EmployeeCount               147
EmployeeNumber              147
EnvironmentSatisfaction     147
Gender                      147
HourlyRate                  147
JobInvolvement              147
JobLevel                    147
JobRole                     147
JobSatisfaction             147
MaritalStatus               147
MonthlyIncome               147
MonthlyRate                 147
NumCompaniesWorked          147
Over18                      147
OverTime                    147
PercentSalaryHike           147
PerformanceRating           147
RelationshipSatisfaction    147
StandardHours               147
StockOptionLevel            147
TotalWorkingYears           147
TrainingTimesLastYear       147
WorkLifeBalance             147
YearsAtC

In [9]:

missing_values = dataset.isnull().sum()
total_values = dataset.shape[0]

percentage_missing = (missing_values / total_values) * 100

print(percentage_missing)

Age                         10.0
Attrition                   10.0
BusinessTravel              10.0
DailyRate                   10.0
Department                  10.0
DistanceFromHome            10.0
Education                   10.0
EducationField              10.0
EmployeeCount               10.0
EmployeeNumber              10.0
EnvironmentSatisfaction     10.0
Gender                      10.0
HourlyRate                  10.0
JobInvolvement              10.0
JobLevel                    10.0
JobRole                     10.0
JobSatisfaction             10.0
MaritalStatus               10.0
MonthlyIncome               10.0
MonthlyRate                 10.0
NumCompaniesWorked          10.0
Over18                      10.0
OverTime                    10.0
PercentSalaryHike           10.0
PerformanceRating           10.0
RelationshipSatisfaction    10.0
StandardHours               10.0
StockOptionLevel            10.0
TotalWorkingYears           10.0
TrainingTimesLastYear       10.0
WorkLifeBa

We can notice that the 10% of each features have missing variable. his is a valid point because it means that we canot drop the missing value due that the dataset will bbe left with really low observation 

In [10]:
dataset.isnull().sum().sum()

5145

total of missing variable are 5145

In [ ]:
Visualization of